skyportal/handlers/api/news_feed.py

```diff
@@ -61,13 +61,52 @@ def fetch_newest(model):
                 .all()
             )
 
+        def fetch_newest_comments():
+            owned_comments = (
+                Comment.query.filter(
+                    Comment.obj_id.in_(
+                        DBSession()
+                        .query(Source.obj_id)
+                        .filter(
+                            Source.group_id.in_(
+                                [g.id for g in self.current_user.accessible_groups]
+                            )
+                        )
+                    )
+                )
+                .order_by(desc(Comment.created_at or Comment.saved_at))
+                .limit(n_items)
+                .all()
+            )
+
+            for comment in owned_comments:
+                author = User.query.filter(User.username == comment.author).first()
+                if author:
+                    comment.author_info = {
+                        "username": author.username,
+                        "first_name": author.first_name,
+                        "last_name": author.last_name,
+                        "gravatar_url": author.gravatar_url,
+                    }
+                else:
+                    unknown_user = User(username="unknown user")
```

Comment:
Could you explain what this section is for?

Reply:
As far as I can tell, the unknown user parts shouldn't really effect production environments - I put them in mainly to deal with test data. Whenever we generate test sources, they get test comments added using the CommentFactory in tests/fixtures.py. But these test comments aren't associated with any users (just given the author "test factory" but that's not a real user) so I was failing all the tests related to this since the API couldn't fetch the user info for those comments. But now that I'm reflecting on this, perhaps the proper solution is to add a User with username "test factory" as part of the base testing database instead?

Reply:
I thought about this some more and I think it would be fine to keep this in instead of adding a "test factory" user to the tests. I can imagine cases in which a user account is deleted and their comments still alive in the database since there is no true foreign key relationship between them (which makes sense I think), so we would want to make sure rendering those orphaned comments doesn't break anything.

Reply:
@acrellin Is that a scenario we hope to support?  It adds a lot of duplicate logic to this PR.

Reply:
I agree that we shouldn't patch the back-end to cover these test-only cases, but should rather update the test fixtures to mimic real-life production DB state. I don't think we'll ever want to support unknown users (deactivated users will still have an associated record).

Reply:
Okay, I've just pushed a commit to remove that logic and just add in a test factory user to the test DB state instead.